### PR TITLE
:sparkles: Add job config to run kubebuilder e2e tests on kubernetes version v1.18.0

### PR DESCRIPTION
**Pull request by prafull01**
Add a kubebuilder job to run e2e tests on kubernetes v1.18.0 version.

Fixes [#1743](https://github.com/kubernetes-sigs/kubebuilder/issues/1743) on kubebuilder

### CODE REVIEW

**Review by camilamacedo86 — APPROVED** [review]
/approve

**Comment by spiffxp** [comment]
how can we grow your group of approvers to more active contributors?

**Comment by spiffxp** [comment]
/cc @DirectXMan12 @pwittrock 
That's all approvers for kubebuilder assigned to this PR

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
I will approve to unblock this time, but backstopping that you have inactive approvers is not our responsibility.  Please address that going forward.

**Comment by alexeldeib** [comment]
@spiffxp thanks for your help, I mentioned this recently in slack. I think part of the problem is all the kubebuilder projects are in kubernetes-sigs, and not all of the reviewers/approvers over there were k8s org members when we first added the jobs + owners: https://github.com/kubernetes/test-infra/pull/17241#issuecomment-614981838

That might be in better shape now, maybe we can add a few more folks. I'm pretty responsive for kubebuilder test-infra even though I haven't been working a ton on KB itself, maybe I can bump myself to approver here for those.

@camilamacedo86 I think you should probably be an approver here at least, lmk if you needs k8s org sponsorship or anything.

**Pull request merged**
